### PR TITLE
Implemented SA1205 (incl. codefix + tests)

### LINKED ISSUE #65: SA1205: PartialElementsMustDeclareAccess
http://www.stylecop.com/docs/SA1205.html

### PR DESCRIPTION
**Pull request by vweijsters**
Fixes #65

### CODE REVIEW

File: StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1205CodeFixProvider.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        {
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+
+        /// <inheritdoc/>
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            foreach (Diagnostic diagnostic in context.Diagnostics.Where(d => FixableDiagnostics.Contains(d.Id)))
+            {
+                context.RegisterCodeFix(CodeAction.Create("Add public access modifier", token => GetTransformedDocumentAsync(context.Document, diagnostic, SyntaxKind.PublicKeyword, token)), diagnostic);
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1205CodeFixProvider.cs]: :exclamation: The symbol will already have a correct access modifier (if no other fragment has a modifier, the default is `internal`). You'll need to get the declared symbol for the class and determine the effective access modifier from that symbol.

Comment (vweijsters) [inline StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1205CodeFixProvider.cs]: Should we go the extra mile and scan all `DeclaringSyntaxReferences` of the symbol? That way we can offer the option to declare the partial element public if none of the other declarations has an access modifier specified, which is more friendly than defaulting to internal.

There might be performance issues associated with it though. Any thoughts?

Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1205CodeFixProvider.cs]: Probably not. We don't do that with other rules like SA1400 (AccessModifierMustBeDeclared).

**Comment by vweijsters** [comment]
Implemented new code fix strategy.

**Pull request merged**